src/videojs-hls.js

```diff
@@ -950,6 +954,7 @@ videojs.Hls.prototype.drainBuffer = function(event) {
   // the playlist
   if (this.duration() !== Infinity && mediaIndex + 1 === playlist.segments.length) {
     this.mediaSource.endOfStream();
+    this.lastSeekedTime_ = undefined;
```

Comment:
Sorry, you're right, that's both the correct behavior and what is actually happening. But null != undefined and still affects the reset at post-end playback.